stacks-and-queues/lib/fifo_animal_shelter.js

```diff
@@ -0,0 +1,58 @@
+'use strict';
+
+const Queue = require('../lib/queue');
+
+const _validate = Symbol('validate');
+
+class AnimalShelter extends Queue {
+  constructor() {
+    super();
+  }
+
+  enqueue(value) {
+    this[_validate](value);
+
+    super.enqueue(value);
+  }
+
+  [_validate](value) {
+    if (typeof value !== 'string') {
+      throw new Error('Input Error: enqueue requires a string "cat" or "dog"');
+    }
+
+    value.toLowerCase();
+    if (value === 'dog' || value === 'cat') {
+      return;
+
+    } else {
+      throw new Error('Input Error: enqueue requires a string "cat" or "dog"');
+    }
+
+    /* 
+    * Any ideas why this condition would resolve to true and throw my error
+    * with value 'cat'?
+    * 
+    * if (value !== 'dog' || value !== 'cat') {

```

Comment:
The first part of your condition says "if the value is not 'dog'" then throw an error. Well, if you're input is 'cat', of course it will throw an error at this first part. The second part will not even be evaluated. It would work if your condition uses && instead.
```
if (value !== 'dog' && value !== 'cat') {}
```